pkg/controller/pravegacluster/upgrade.go

```diff
@@ -315,6 +315,9 @@ func (r *ReconcilePravegaCluster) checkUpdatedPods(sts *appsv1.StatefulSet, vers
 
 		if !util.IsPodReady(pod) {
 			// At least one updated pod is still not ready
+			if pod.Status.ContainerStatuses[0].State.Waiting.Reason == "ImagePullBackOff" {
```

Comment:
If the container starts correctly(without any errors), there would not be any value like `pod.Status.ContainerStatuses[0].State.Waiting` and this code would throw an exception.
It would be good to have a check like:
` if pod.Status.ContainerStatuses[0].State == ContainerState.Waiting 
 && pod.Status.ContainerStatuses[0].State.Reason == ImagePullBackOff`

Reply:
If the container starts correctly, `pod.Status.ContainerStatuses[0].State.Waiting` is set to nil. Added the following check to the code to avoid such exceptions :
`if pod.Status.ContainerStatuses[0].State.Waiting != nil && pod.Status.ContainerStatuses[0].State.Waiting.Reason == "ImagePullBackOff"`